### ISSUE TITLE
floods logs with empty lines etc

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#5386]
This is a random selection from a travis run for reproman https://travis-ci.org/github/ReproNim/reproman/jobs/756520237 for https://github.com/ReproNim/reproman/pull/567

```
2021-01-28 12:27:24,815 [INFO   ] Creating run commit in /tmp/pytest-of-travis/pytest-0/dataset25 
Total: 0.00 datasets [00:00, ? datasets/s]
Total:   0%|                                   | 0.00/6.12k [00:00<?, ? Bytes/s]
.
p/pytest-of-travis/pytest-0/dataset26 
0.00 [01:53, ?/s]
0.00 [01:37, ?/s]
```

where we have 
```
Collecting datalad>=0.13.0rc2
  Downloading datalad-0.14.0rc1.tar.gz (1.6 MB)
```
(I will now remove rc from our dependencies version there and see if it goes away with released version)

It might still be an issue somehow of how datalad orchestrators invoke it but AFAIK they do not do anything special for logging etc (but may be they do)

**Comment by yarikoptic** [comment]
Similar situation is with heudiconv which floods the screen with all the 
```
INFO: Resolving deltas:  65% (20373/31300)
INFO: Resolving deltas:  66% (20709/31300)
```
while running tests.

I think at large there is an aspect of us (datalad) doing the "wrong thing" by setting up formatters and handlers for the logs upon import regardless if datalad is used as an application (e.g. via CLI) or used within a bigger app as a python module.  I believe the common practice is to not setup logging formatters etc if module is used as a part of some bigger application since it would be up for that application to do that to its liking.  So may be we should RF our log.py so `lgr` by default is not getting initialized and then only initialize it whenever used by CLI or in setup_package for tests

**Comment by kyleam** [comment]
> think at large there is an aspect of us (datalad) doing the "wrong thing" [...]

I agree with that [1], but...

> So may be we should RF our log.py so lgr by default is not getting initialized and then only initialize it whenever used by CLI or in setup_package for tests

... without a major change in design, wouldn't this be dropping progress bars (and other useful information) for Python use of datalad, particularly interactive use?  It seems like there's a conflict between "make datalad (the application) work as close as possible from the python and command-line interfaces" and "make datalad a good library".

[1] I know we've discussed this before, but perhaps most of that was offline.  The only relevant online discussion I can find is <https://github.com/datalad/datalad/pull/5104#issuecomment-718830958>.

**Comment by mih** [comment]
I agree with this assessment. I think at present, and without a clear concept under which conditions we consider "library-type use" vs "application-type use" there is nothing we can reasonably do here.
